File: 双指针/11.盛雨水.java
class Solution {
    public int maxArea(int[] height) {
        int i=0,j=height.length-1;
        int res=0;
        while(i<j){
            int h=0;
            if(height[i]<height[j]){
                h=height[i];
                i++;
            }else{
                h=height[j];
                j--;
            }
            int s=(j-i+1)*h;
            res=Math.max(res,s);
        }
        return res;
    }
}

File: 双指针/11.盛雨水.go
package 双指针

func maxArea(height []int) int {
	l, r := 0, len(height)-1
	res := 0
	for l < r {
		h := 0
		if height[l] < height[r] {
			h = height[l]
			l++
		} else {
			h = height[r]
			r--
		}
		s := (r - l + 1) * h
		res = max(res, s)
	}
	return res
}
